optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -698,3 +828,114 @@ def _reorder_cache(past, beam_idx) -> Tuple[Tuple[torch.FloatTensor]]:
                 tuple(past_state.index_select(0, beam_idx) for past_state in layer_past[:2]) + layer_past[2:],
             )
         return reordered_past
+
+
+class ORTModelForSpeechSeq2Seq(ORTModelForConditionalGeneration, GenerationMixin):
+    """
+    Speech Sequence-to-sequence model with a language modeling head for ONNX Runtime inference.
+    """
+
+    export_feature = "speech2seq-lm"
+    auto_model_class = AutoModelForSpeechSeq2Seq
+    main_input_name = "input_features"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _initialize_encoder(
+        self, encoder_session: onnxruntime.InferenceSession, device: str
+    ) -> ORTEncoderForSpeechSeq2Seq:
+        return ORTEncoderForSpeechSeq2Seq(session=encoder_session, device=device, main_input_name=self.main_input_name)
+
+    def get_encoder_onnx_config(encoder_config: PretrainedConfig) -> OnnxConfig:
+        return SpeechSeq2SeqEncoderOnnxConfig(encoder_config, task="default")
```

Comment:
Currently,  the class `ORTModelForSpeechSeq2Seq` returns generic encoder / decoder ONNX configs. Same design is for the `ORTModelForSeq2SeqLM` class. 

`AutoModelForSpeechSeq2Seq` contains multiple speech model classes within it. There may be some models which are not supported using the generic ONNX configs (Need to confirm)? Similar to AutoClasses we may need a structure to switch to appropriate Onnx Config based on the model types. Probably something we can work after refactor.

Reply:
Since the whole ONNX part is also changing, I think it's fine to allow time for a refactor to support audio if needed